scipy/sparse/tests/test_base.py

```diff
@@ -758,19 +758,22 @@ def test_sum(self):
         def check(dtype, j):
             dat = np.matrix(matrices[j], dtype=dtype)
             datsp = self.spmatrix(dat, dtype=dtype)
-            assert_array_almost_equal(dat.sum(), datsp.sum())
-            assert_equal(dat.sum().dtype, datsp.sum().dtype)
-            assert_(np.isscalar(datsp.sum(axis=None)))
-            assert_array_almost_equal(dat.sum(axis=None), datsp.sum(axis=None))
-            assert_equal(dat.sum(axis=None).dtype, datsp.sum(axis=None).dtype)
-            assert_array_almost_equal(dat.sum(axis=0), datsp.sum(axis=0))
-            assert_equal(dat.sum(axis=0).dtype, datsp.sum(axis=0).dtype)
-            assert_array_almost_equal(dat.sum(axis=1), datsp.sum(axis=1))
-            assert_equal(dat.sum(axis=1).dtype, datsp.sum(axis=1).dtype)
-            assert_array_almost_equal(dat.sum(axis=-2), datsp.sum(axis=-2))
-            assert_equal(dat.sum(axis=-2).dtype, datsp.sum(axis=-2).dtype)
-            assert_array_almost_equal(dat.sum(axis=-1), datsp.sum(axis=-1))
-            assert_equal(dat.sum(axis=-1).dtype, datsp.sum(axis=-1).dtype)
+            with np.errstate(over='ignore'):
```

Comment:
If you only want to catch the tests that throw overflow warnings in the context manager, you need something like this:
````
# Ignore the overflow warnings that result from casting negative
# numbers to unsigned integers and then using the dok_matrix
# implementation of matrix * vector multiplication:
with np.errstate(over='ignore' if (
        np.any(matrices[j] < 0) and
        np.dtype(dtype).kind == 'u' and
        datsp.getformat() == 'dok' )
                 else 'warn'):
    ...
````
This might be overkill, though. Maybe I could leave it as it is, and just add the comment? An alternative is to remove the test cases in which negative numbers are cast to unsigned integers. That would assume that those cases do not need to behave consistently with NumPy. If the casting behavior from int to uint is assumed to be defined, however, then removal of those cases is also not desirable, I think.